### DISCUSSION TITLE
Alternatives to OpenOPC

### DISCUSSION THREAD

**Discussion by renzop** [iterativ/openopc2#53] [Show and tell]
Use this discusion to talk about other libraries that can be used for similar purposes as Openopc2. This thread has been opened to avoid "spamming" (hey I made this library ... ) in every other issue or discussion. Therefore advertise your Project here if you like.

**Comment by duduyoyo** [comment]
If you feel pain in configuring DCOM between OPC Client and Server, or can't harden your wrapper class due to no access to source code, you really check this [solution](https://github.com/duduyoyo/WebSocket4OPC) out.

Simply put this solution can help you retrieve OPC DA/AE/HDA data back without worry using DCOM or DCOM hardening. So it is future-proof for any future Microsoft DCOM vulnerability. Architecturally it is designed to work in such a modern way so you can pick up your favorite language（C/C#/Java/JavaScript/Python/Swift）and receive data in your familiar JSON format. No gateway service installation required and support cross-platform in various clients (desktop/mobile/cloud/edge). Hope it can help protect your investment in Classic OPC for a long time to come.
